_posts/2022/2022-10-31-use-zero-for-integer-float-comparison.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Use zero? for comparison of numerics like Integer, Float and BigDecimal"
+description: "Other languages don't have this"
+layout: article
+category: ruby
+image:
+  base: "2022/use-zero-for-integer-float-comparison"
+  alt: "A zero on a sign on the wall"
+  credit: "Bernard Hermant"
+  source: "https://unsplash.com/photos/rfpSOlH1JlQ"
+---
+
+Ruby often takes a more “readable” approach to its syntax and standard library of included methods than other languages.
+
+An example of this is the common programming comparison of numeric variable to zero.
+
+See the documentation of the `.zero?` method on [`Integer`](https://ruby-doc.org/core-3.1.2/Integer.html#zero-3F-method), [`Float`](https://ruby-doc.org/core-3.1.2/Float.html#zero-3F-method), [`Numeric`](https://ruby-doc.org/core-3.1.2/Numeric.html#zero-3F-method) and [`BigDecimal`](https://ruby-doc.org/stdlib-3.1.2/libdoc/bigdecimal/rdoc/BigDecimal.html#zero-3F-method).
+
+
+## Instead of...
+
+...checking whether a value is zero in a conditional, how other languages might do it:
+
+```ruby
+if number == 0
+  :yes
+else
+  :no
+end
+```
+
+
+## Use...
+
+...the built-in `#zero?` method on all numeric types:
+
+```ruby
+if number.zero?
+  :yes
+else
+  :no
+end
+```
+
+
+## Why?
+
+The Ruby-ish syntax in this case is clearer and less error prone.
+
+It is very easy to type `number = 0` and thus assign a variable in the conditional rather than compare (although you might use tests to protect yourself from this particular error).
+
+
+## Why not?
+
+This is a habit I still struggle with after 15 years of Ruby programming. The `== 0` syntax is how _many_ other similar languages do comparison and is deeply ingrained in my fingers!
```

Comment:
Maybe lead with the issue of how many other languages do it...and then say "Ever after 15 years of Ruby programming..."

Reply:
```suggestion
The `== 0` syntax is how _many_ other similar languages perform comparison and is deeply ingrained in my fingers. I still struggle to apply this advice even after 15 years of Ruby programming.
```